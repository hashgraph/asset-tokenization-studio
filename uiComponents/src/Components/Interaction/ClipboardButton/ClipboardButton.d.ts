/// <reference types="react" />
import type { BaseMultiStyleConfiguration } from "@Theme/Components/BaseMultiStyleConfiguration";
import type { IconButtonProps } from "@Components/Interaction/IconButton";
export declare const clipboardButtonPartsList: Array<"iconButton" | "icon" | "tooltip">;
type Parts = typeof clipboardButtonPartsList;
export interface ClipboardButtonConfiguration extends BaseMultiStyleConfiguration<Parts> {
}
export interface ClipboardButtonProps extends Omit<IconButtonProps, "as" | "onClick" | "aria-label"> {
    value: string;
    variant?: string;
    size?: string;
}
export declare const ClipboardButton: ({ value, variant, size, ...props }: ClipboardButtonProps) => JSX.Element;
export {};
