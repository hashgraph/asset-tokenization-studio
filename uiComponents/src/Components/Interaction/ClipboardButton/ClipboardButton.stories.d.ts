declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./ClipboardButton").ClipboardButtonProps>;
export default meta;
export declare const WithoutLabelTooltip: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./ClipboardButton").ClipboardButtonProps>;
