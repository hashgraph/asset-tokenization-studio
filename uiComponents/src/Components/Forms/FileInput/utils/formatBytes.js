export var formatBytes = function (bytes) {
    if (bytes === 0)
        return "0 Bytes";
    var k = 1024;
    var sizes = ["Bytes", "KB", "MB", "GB", "TB"];
    var i = Math.floor(Math.log(bytes) / Math.log(k));
    return parseFloat((bytes / Math.pow(k, i)).toFixed(2)) + " " + sizes[i];
};
