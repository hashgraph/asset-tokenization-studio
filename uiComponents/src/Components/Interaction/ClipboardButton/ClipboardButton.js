var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
import { CopySimple } from "@phosphor-icons/react";
import { IconButton } from "@Components/Interaction/IconButton";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { useClipboard } from "@chakra-ui/react";
export var clipboardButtonPartsList = ["iconButton", "icon", "tooltip"];
export var ClipboardButton = function (_a) {
    var value = _a.value, variant = _a.variant, _b = _a.size, size = _b === void 0 ? "xxs" : _b, props = __rest(_a, ["value", "variant", "size"]);
    var styles = useChakraMultiStyleConfig("ClipboardButton", { variant: variant });
    var onCopy = useClipboard(value).onCopy;
    return (React.createElement(IconButton, __assign({ "aria-label": "Copy to clipboard-".concat(value), icon: React.createElement(PhosphorIcon, { sx: styles.icon, size: size, as: CopySimple }), onClick: function (e) {
            e.stopPropagation();
            onCopy();
        }, variant: "tertiary", sx: styles.iconButton }, props)));
};
