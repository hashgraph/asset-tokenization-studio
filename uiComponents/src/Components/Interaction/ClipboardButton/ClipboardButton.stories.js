var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Box } from "@chakra-ui/react";
import { ClipboardButton } from "./ClipboardButton";
var meta = {
    title: "Design System/Interaction/ClipboardButton",
    component: ClipboardButton,
    argTypes: {
        value: {
            control: {
                type: "text"
            },
            description: "Text to copy to clipboard"
        },
        label: {
            control: {
                type: "text"
            },
            description: "Label to show when copy to clipboard"
        }
    },
    parameters: {
        docs: {}
    },
    args: {
        value: "Copied from clipboard",
        label: ""
    }
};
export default meta;
var Template = function (args) { return (React.createElement(Box, { maxWidth: "50px" },
    React.createElement(ClipboardButton, __assign({}, args)))); };
export var WithoutLabelTooltip = Template.bind({});
WithoutLabelTooltip.args = {};
