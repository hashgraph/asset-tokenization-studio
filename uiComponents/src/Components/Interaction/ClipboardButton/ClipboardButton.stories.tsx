import React from "react";
import { Box } from "@chakra-ui/react";
import type { Meta, StoryFn } from "@storybook/react";
import { ClipboardButton } from "./ClipboardButton";

const meta = {
  title: "Design System/Interaction/ClipboardButton",
  component: ClipboardButton,
  argTypes: {
    value: {
      control: {
        type: "text",
      },
      description: "Text to copy to clipboard",
    },
    label: {
      control: {
        type: "text",
      },
      description: "Label to show when copy to clipboard",
    },
  },
  parameters: {
    docs: {},
  },
  args: {
    value: "Copied from clipboard",
    label: "",
  },
} as Meta<typeof ClipboardButton>;
export default meta;

const Template: StoryFn<typeof ClipboardButton> = (args) => (
  <Box maxWidth="50px">
    <ClipboardButton {...args} />
  </Box>
);

export const WithoutLabelTooltip = Template.bind({});
WithoutLabelTooltip.args = {};
