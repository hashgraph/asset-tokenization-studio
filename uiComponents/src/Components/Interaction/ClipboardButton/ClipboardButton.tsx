import React from "react";
import { useMultiStyleConfig as useChakraMultiStyleConfig } from "@chakra-ui/system";
import type { BaseMultiStyleConfiguration } from "@Theme/Components/BaseMultiStyleConfiguration";
import { CopySimple } from "@phosphor-icons/react";
import type { IconButtonProps } from "@Components/Interaction/IconButton";
import { IconButton } from "@Components/Interaction/IconButton";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { useClipboard } from "@chakra-ui/react";

export const clipboardButtonPartsList: Array<
  "iconButton" | "icon" | "tooltip"
> = ["iconButton", "icon", "tooltip"];

type Parts = typeof clipboardButtonPartsList;

export interface ClipboardButtonConfiguration
  extends BaseMultiStyleConfiguration<Parts> {}

export interface ClipboardButtonProps
  extends Omit<IconButtonProps, "as" | "onClick" | "aria-label"> {
  value: string;
  variant?: string;
  size?: string;
}

export const ClipboardButton = ({
  value,
  variant,
  size = "xxs",
  ...props
}: ClipboardButtonProps) => {
  const styles = useChakraMultiStyleConfig("ClipboardButton", { variant });

  const { onCopy } = useClipboard(value);

  return (
    <IconButton
      aria-label={`Copy to clipboard-${value}`}
      icon={<PhosphorIcon sx={styles.icon} size={size} as={CopySimple} />}
      onClick={(e) => {
        e.stopPropagation();
        onCopy();
      }}
      variant="tertiary"
      sx={styles.iconButton}
      {...props}
    />
  );
};
