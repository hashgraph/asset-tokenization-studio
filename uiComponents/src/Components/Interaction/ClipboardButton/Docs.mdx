import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as ClipboardButtonStories from "./ClipboardButton.stories";

<Meta of={ClipboardButtonStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/ioBricks-Design-System?type=design&node-id=1503%3A23013&mode=design&t=KlerdyEMWSQWLAVX-1"
  target="\_blank"
>
  View figma design
</a>

<Primary />

## How to set up

### Config theme

Add the following configurations to the `theme.tsx` and `ClipboardButton.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    ClipboardButton: ClipboardButtonTheme,
  },
};
```

```tsx dark
const ClipboardButtonTheme: ClipboardButtonConfiguration = {
  parts: clipboardButtonPartsList,
  baseStyle: {},
};
```

### Component parts

You have these parts available: `"iconButton" | "icon" | "tooltip"` (`clipboardButtonPartsList`)

## Props

<ArgTypes of={ClipboardButtonStories} />

## Stories

<Stories title="" />
