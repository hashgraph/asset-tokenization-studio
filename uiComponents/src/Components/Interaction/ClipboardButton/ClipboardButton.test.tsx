import React from "react";
import {} from "@chakra-ui/utils";
import { render } from "@/test-utils";
import { fireEvent } from "@testing-library/react";
import type { ClipboardButtonProps } from "./ClipboardButton";
import { ClipboardButton } from "./ClipboardButton";

const onCopyMock = jest.fn();
jest.mock("@chakra-ui/react", () => ({
  ...jest.requireActual("@chakra-ui/react"),
  useClipboard: jest.fn(() => ({
    hasCopied: false,
    onCopy: onCopyMock,
  })),
}));

describe(`< ${ClipboardButton.name}/>`, () => {
  const defaultProps: ClipboardButtonProps = {
    value: "Hola mundo",
  };

  const factoryComponent = (props: Partial<ClipboardButtonProps> = {}) => {
    return render(<ClipboardButton {...defaultProps} {...props} />);
  };

  it("should render without crashing", () => {
    const { asFragment } = factoryComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should copy to clipboard", () => {
    const { getByRole } = factoryComponent();
    const button = getByRole("button");
    fireEvent.click(button);
    expect(onCopyMock).toHaveBeenCalledTimes(1);
  });
});
