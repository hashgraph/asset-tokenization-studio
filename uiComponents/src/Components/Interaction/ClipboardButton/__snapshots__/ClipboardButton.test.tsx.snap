// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ClipboardButton/> should render without crashing 1`] = `
<DocumentFragment>
  <button
    aria-label="Copy to clipboard-Hola mundo"
    class="chakra-button css-jwxhfr"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-kpbtv0"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 256 256"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M184,64H40a8,8,0,0,0-8,8V216a8,8,0,0,0,8,8H184a8,8,0,0,0,8-8V72A8,8,0,0,0,184,64Zm-8,144H48V80H176ZM224,40V184a8,8,0,0,1-16,0V48H72a8,8,0,0,1,0-16H216A8,8,0,0,1,224,40Z"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
